packages/odyssey-react/src/components/Infobox/index.tsx

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from 'react';
+import type { FunctionComponent, ReactNode } from 'react';
+import { useCx, useOmit } from '../../utils';
+
+export type InfoboxVariants = 'info' | 'danger' | 'caution' | 'success';
+
+export type Props = {
+  /**
+   * Content to be rendered within the infobox. Avoid using direct children, put child content
+   * within the provided Infobox static components (Infobox.Content and Infobox.Actions)
+   */
+  children: ReactNode,
+  
+  /**
+   * The visual variant to be displayed to the user.
+   * @default info
+   */
+  variant?: InfoboxVariants,
+
+  /**
+   * The title or headline of the Infobox. If Infobox.Content is not present it is required.
+   */
+  title?: string;
+};
+
+type PropsInfoboxContent = {  
+  children: ReactNode
+}
+
+type PropsInfoboxActions = {  
+  children: ReactNode
+}
+
+export type StaticComponents = {
+  Content: FunctionComponent<PropsInfoboxContent>,
+  Actions: FunctionComponent<PropsInfoboxActions>,
+}
+
+/**
+ * An infobox is a type of alert that provides feedback in response to a 
+ * user action or system activity.
+ *
+ * @component
+ * @example
+ * <Infobox variant="primary" title="Moonbase Alpha-6">
+ *  <Infobox.Content>
+ *    You are currently logged in from Moonbase Alpha-6, located on Luna.
+ *  </Infobox.Content>
+ *  <Infobox.Actions>
+ *    <Link href="/fuel">Visit fueling console</Link>
+ *  </Infobox.Actions>
+ * </Infobox>
```

Comment:
Let me know if I'm understanding this @jeffbelser-okta… Are you suggesting that this allows developers to re-order these elements so `Infobox.Actions` can be positioned above `Infobox.Content`? If so, we wouldn't want to allow that.

Reply:
@joejezowski-okta yes the order can be changed between actions and content or other arbitrary UI could be rendered within the Infobox container. I'm not familiar enough with what Infobox is trying to accomplish to know if this is good or not?

Composition based APIs also exist for `Modal` and `Radio.Group` as well. `Modal` makes sense because the content will necessarily be dynamic - `Radio Group` perhaps less so but there could be use cases where radio inputs are split between other UI or scenarios where a consumer wants to attach specific attributes or event handlers to individual inputs.

In general I'm in favor of allowing our components to be flexible via composition. However, when we absolutely *need* to enforce restrictions on how they can be used we should have the conversation about risk vs reward for the effort it requires to enforce those rules in code and not just through documentation or convention.

The other issue for us to consider is the complexity with composition based APIs. Is requiring the user to place an `Infobox.Content` and `Infobox.Actions` component as children of `Infobox` adding value? Or simply requiring a more verbose and error prone way of adding text into a component template we want to control?

Reply:
This is a really good piece of feedback @jeffbelser-okta. I'll admit I didn't really think about this in the case of Infobox or Modal. I incorrectly made an assumption that our developers wouldn't incorrectly compose an infobox because a designer won't incorrectly design an infobox.

This kind of question is also why I want us to define API design standards so that the implementation isn't up to the developer's discretion but rather a set of guidelines.

Would you feel comfortable with us re-evaluating this in our API sweep, or would you prefer we fix this now? If the latter, my mind immediately goes to implementing the component this way:

```html
<Infobox
  title="Infobox title"
  content={() => (
    <>
      <Text>Some infobox content here.</Text>
      <Text>A little more content.</Text>
    </>
  )}
  actions={() => (
      <Link href="https://www.okta.com">Action link.</Link>
  )}
/>
``` 

I have never loved this type of component API interface, but it _does_ solve the problem. Any other ideas?

Reply:
talked in OH but lets keep what we have with the understanding that while it does introduce more flexibility than is strictly necessary, it also is consistent against other component APIs we're building for Modal, Table, Radio etc.